### PR TITLE
feat: adicionando birthdate e adoption_date ao UPDATE do pet   -  issue56

### LINKED ISSUE #56: [UPDATE PET] Incluir campo propriedade Aniversario e Dia de doacao no endpoint de Update Pet
**Descreva a funcionalidade**
Atualizar o endpoint de atualizar informações de um animal (Pet), permitindo que as propriedades "Aniversário" e "Dia de Doação" sejam recebidas e atualizadas.
Atualizar os campos que nao existem no banco via migrations

**Descrição**
Essa funcionalidade tem como objetivo possibilitar que os usuários atualizem as informações relacionadas às datas importantes de um animal, como seu aniversário e o dia em que foi adotado (Dia de Doação). Isso permite manter registros precisos e significativos sobre a história do animal.

**Cenário de Uso**
A inclusão da capacidade de atualizar as propriedades "Aniversário" e "Dia de Doação" é útil quando os usuários desejam manter informações detalhadas sobre os eventos significativos na vida do animal.

**Detalhes de Implementação**
A implementação envolve a expansão do endpoint de atualização de pets para aceitar uma requisição que inclua as novas datas. A lógica no backend deve validar e processar a requisição, atualizando as propriedades "Aniversário" e "Dia de Doação" do animal no banco de dados.

**Exemplo de Requisição**
```json
PATCH /api/pets/{id}
{
  "aniversario": "2023-05-15",
  "dia_doacao": "2023-08-01"
}
```

**Conceitos Visuais**
[Incluir qualquer esboço ou representação visual, se aplicável.]

**Informações Adicionais**
A documentação do endpoint deve ser atualizada para refletir essa nova capacidade, e os desenvolvedores e usuários finais devem ser informados sobre a existência e uso desse recurso. Testes abrangentes devem ser realizados para garantir o correto funcionamento do endpoint de atualização do pet com a inclusão das novas propriedades.

### PR DESCRIPTION
**Pull request by yuri-silveiraa**
Closes #56 

<details open> 
  <summary>
    <b>adicionando campos birthdate e adoption_date ao UPDATE do pet</b>
  </summary>

N/A
</details>

<details open> 
  <summary>
    <b>Bugfix</b>
  </summary>

- **Description**
N/A

- **Cause**
N/A

- **Solution**
N/A
</details>

<details> 
  <summary>
    <b>Changelog</b>
  </summary>
N/A
</details>

<details open> 
  <summary>
    <b>Visual evidences :framed_picture:</b>
  </summary>

</details>

<details open> 
  <summary>
    <b>Checklist</b>
  </summary>

  - [ ] Issue linked
  - [x] Build working correctly
  - [x] Tests created
</details>

<details> 
  <summary>
    <b>Additional info</b>
  </summary>
N/A
</details>

### CODE REVIEW

File: usecase/pet.go [hunk 1]
```diff
@@ -30,21 +30,26 @@ func (c *PetUseCase) FindByID(ID uniqueEntityId.ID) (*entity.Pet, error) {
 func (c *PetUseCase) Update(petID string, userID string, petUpdateDto dto.PetUpdatetDto) (err error) {
 	petToUpdate := petUpdateDto.ToEntity()
 
+	fmt.Println(petToUpdate.Size + "")
```
Comment (kvnbg) [inline usecase/pet.go:33]: remover esse "log"

File: usecase/pet.go [hunk 2]
```diff
@@ -30,21 +30,26 @@ func (c *PetUseCase) FindByID(ID uniqueEntityId.ID) (*entity.Pet, error) {
...
 	if !c.isValidPetSize(petToUpdate) {
 		return errors.New("the animal size is invalid")
 	}
 
 	err = c.repo.Update(petID, userID, petToUpdate)
 	if err != nil {
-		return fmt.Errorf("failed to update size for pet with ID %s: %w", petID, err)
+		return fmt.Errorf("failed to update for pet with ID %s: %w", petID, err)
```
Comment (kvnbg) [inline usecase/pet.go:40]: Quando fizer o rebase com main, adiciona também a linha de log informando o erro

**Review by kvnbg — CHANGES_REQUESTED** [review]
Fazer o rebase com a main para resolver os conflitos, teve um merge recente na main

File: usecase/pet.go [hunk 3]
```diff
@@ -30,21 +30,26 @@ func (c *PetUseCase) FindByID(ID uniqueEntityId.ID) (*entity.Pet, error) {
...
 
 	return nil
 }
 
 func (c *PetUseCase) isValidPetSize(petToUpdate *entity.Pet) bool {
-	return &petToUpdate.Size != nil && petToUpdate.Size != "" &&
-		(petToUpdate.Size == "small" || petToUpdate.Size == "medium" || petToUpdate.Size == "large" || petToUpdate.Size == "giant")
+    var size = petToUpdate.Size
+
+    if size != "" {
```
Comment (devlefel) [inline usecase/pet.go:49]: Pode retirar essa parte a @mmaysc ja ajustou

**Review by devlefel — CHANGES_REQUESTED** [review]

**Review by devlefel — APPROVED** [review]

**Review by devlefel — APPROVED** [review]

**Review by kvnbg — APPROVED** [review]

**Pull request merged**